exercises/rational-numbers/canonical-data.json

```diff
@@ -0,0 +1,356 @@
+          "property": "exprational",
+          "input": {
+            "r": [0, 1],
+            "n": 5
+          },
+          "expected": [0, 1]
+        },
+        {
+          "description": "Raise one to an integer power",
+          "property": "exprational",
+          "input": {
+            "r": [1, 1],
+            "n": 4
+          },
+          "expected": [1, 1]
+        },
+        {
+          "description": "Raise a positive rational number to the power of zero",
+          "property": "exprational",
+          "input": {
+            "r": [1, 2],
+            "n": 0
+          },
+          "expected": [1, 1]
+        },
+        {
+          "description": "Raise a negative rational number to the power of zero",
+          "property": "exprational",
+          "input": {
+            "r": [-1, 2],
+            "n": 0
+          },
+          "expected": [1, 1]
+        }
+      ]
+    },
+    {
+      "description": "Exponentiation of a real number to a rational number",
+      "cases": [
+        {
+          "description": "Raise a real number to a positive rational number",
+          "property": "expreal",
+          "input": {
+            "x": 8,
+            "r": [4, 3]
+          },
+          "expected": 16.0
+        },
+        {
+          "description": "Raise a real number to a negative rational number",
+          "property": "expreal",
+          "input": {
+            "x": 9,
+            "r": [-1, 2]
+          },
+          "expected": 0.3333333333333333
+        },
+        {
+          "description": "Raise a real number to a zero rational number",
+          "property": "expreal",
+          "input": {
+            "x": 2,
+            "r": [0, 1]
+          },
+          "expected": 1.0
+        }
+      ]
+    },
+    {
+      "description": "Reduction to lowest terms",
+      "cases": [
+        {
+          "description": "Reduce a positive rational number to lowest terms",
+          "property": "reduce",
+          "input": [2, 4],
+          "expected": [1, 2]
+        },
+        {
+          "description": "Reduce a negative rational number to lowest terms",
+          "property": "reduce",
+          "input": [-4, 6],
+          "expected": [-2, 3]
+        },
+        {
+          "description": "Reduce a rational number with a negative denominator to lowest terms",
+          "property": "reduce",
+          "input": [3, -9],
+          "expected": [-1, 3]
+        },
+        {
+          "description": "Reduce zero to lowest terms",
+          "property": "reduce",
+          "input": [0, 6],
+          "expected": [0, 1]
+        },
+        {
+          "description": "Reduce an integer to lowest terms",
+          "property": "reduce",
+          "input": [-14, 7],
+          "expected": [-2, 1]
+        },
+        {
+          "description": "Reduce one to lowest terms",
+          "property": "reduce",
+          "input": [13, 13],
```

Comment:
> let's make `input` an object with named keys/values, instead of an array like it is on this line
> 
> the reason to do this would be that:
> 
> * it is consistent with the rest of the cases
> * it is compliant with https://github.com/exercism/problem-specifications/issues/996#issuecomment-342740372

same

Reply:
I see. New change pushed.